internal/controller/telemetry/collector.go

```diff
@@ -189,6 +185,7 @@ func (c DataCollectorImpl) Collect(ctx context.Context) (Data, error) {
 		SnippetsFiltersDirectivesCount: snippetsFiltersDirectivesCount,
 		NginxPodCount:                  nginxPodCount,
 		ControlPlanePodCount:           int64(replicaCount),
+		NginxOneConnectionEnabled:      c.cfg.NginxOneConsoleSecretName != "",
```

Comment:
Rather than passing the Secret all the way to the collector, let's just set this boolean when we initialize it in the manager.